src/pages/Message.jsx

```diff
@@ -0,0 +1,134 @@
+import { Box, Stack, Typography } from "@mui/material";
+import React, { useEffect, useState } from "react";
+import MessagesComp from "../components/MessagesComp";
+import NewMessage from "../components/NewMessage";
+
+import { firestore, auth } from "../firebase";
+import {
+  collection,
+  query,
+  getDocs,
+  doc,
+  updateDoc,
+  where
+} from "firebase/firestore";
+
+function Message() {
+  const [message, setMessage] = useState([]);
+  const [newMessage, setNewMessage] = useState("");
+  const [user, setUser] = useState(null);
+  const [childId, setChildId] = useState(null);
+  const [childRef, setChildRef] = useState(null)
+  const [messageDocRef, setMessageDocRef] = useState(null)
+
+  useEffect(() => {
+    const unsubscribe = auth.onAuthStateChanged((user) => {
+      setUser(user);
+    });
+    return unsubscribe;
+  });
+
+
+  useEffect(() => {
+    const fetchData = async () => {
+      console.log("new page")
+      try {
+        const collectionRef = collection(firestore, "Child");
+        // get users email from =====>>>>> user.email
+        const q = query(collectionRef, where("email", '==', 'penpalprogram.murphycharity@gmail.com'))
```

Comment:
Is it possible to perform this on the backend somehow (e.g., by exchanging a session token)?